src/utils.rs

```diff
@@ -0,0 +1,79 @@
+use std::str::Chars;
```

Comment:
if we're working with byte strings and need UTF-8 decoding (which I suspect we don't if we're only matching ASCII specifiers), then we should pull in `bstr`.